vmdb/app/models/miq_provision_cloud/cloning.rb

```diff
@@ -1,7 +1,15 @@
 module MiqProvisionCloud::Cloning
   def find_destination_in_vmdb(ems_ref)
-    platform = "Vm" + self.class.name.split("MiqProvision").last
-    platform.constantize.where(:ems_id => source.ext_management_system.id, :ems_ref => ems_ref).first
+    vm_model_class.where(:ems_id => source.ext_management_system.id, :ems_ref => ems_ref).first
+  end
+
+  def vm_model_class
+    case self.class.name
+    when "MiqProvisionAmazon"
+      ManageIQ::Providers::Amazon::CloudManager::Vm
+    when "MiqProvisionOpenstack"
+      VmOpenstack
+    end
```

Comment:
Out of curiosity, when we covert openstack, will this go back to being dynamically detected, or were you thinking of keeping it explicit.  I only ask because as we add Azure, this is like another place that has to be known to the developer.  (Or perhaps the MiqProvision class should just declare it's own `vm_model_class`.)